python/sglang/srt/entrypoints/openai/serving_embedding.py

```diff
@@ -71,111 +71,61 @@ def _validate_request(self, request: EmbeddingRequest) -> Optional[str]:
+                        generate_prompts.append(conv.get_prompt())
+                else:
+                    generate_prompts = texts
+
+                if len(generate_prompts) == 1:
+                    prompt_kwargs = {
+                        "text": generate_prompts[0],
+                        "image_data": images[0],
+                    }
                 else:
-                    # List of integers (token IDs) or empty list
-                    prompt_kwargs = {"input_ids": prompt}
+                    prompt_kwargs = {
+                        "text": generate_prompts,
+                        "image_data": images,
+                    }
             else:
-                # Other types (should not happen but handle gracefully)
+                # List of integers (token IDs) or empty list
                 prompt_kwargs = {"input_ids": prompt}
-            # Use the passed request_ids for single request
-            final_request_id = request_ids[0] if len(all_requests) == 1 else request_ids
         else:
-            # Handle batch requests
-            if len(prompts) > 0:
-                # Validate that all prompts have the same type
-                first_prompt = prompts[0]
-                first_type = type(first_prompt)
-                for i, prompt in enumerate(prompts[1:], 1):
-                    if type(prompt) != first_type:
-                        raise AssertionError(
-                            f"All prompts in batch must have the same type, but prompt at index {i} has different type"
-                        )
-
-                if isinstance(first_prompt, str):
-                    # Batch of strings
-                    prompt_kwargs = {"text": prompts}
-                elif isinstance(first_prompt, list):
-                    if len(first_prompt) > 0 and isinstance(first_prompt[0], str):
-                        # Batch of lists of strings
-                        prompt_kwargs = {"text": prompts}
-                    elif len(first_prompt) > 0 and isinstance(
-                        first_prompt[0], MultimodalEmbeddingInput
-                    ):
-                        # Handle multimodal batch requests
-                        raise NotImplementedError(
-                            "Multiple requests with multimodal inputs are not supported yet"
-                        )
-                    else:
-                        # Batch of token ID lists
-                        prompt_kwargs = {"input_ids": prompts}
-                else:
-                    # Other types
-                    prompt_kwargs = {"input_ids": prompts}
-            else:
-                prompt_kwargs = {"input_ids": prompts}
-            # Use the passed request_ids for batch requests
-            final_request_id = request_ids
-
+            # Other types (should not happen but handle gracefully)
+            prompt_kwargs = {"input_ids": prompt}
         adapted_request = EmbeddingReqInput(
-            rid=final_request_id,
```

Comment:
I think this rid is mainly for the log_requests purpose. Can you check what happens if there is batch inputs in the EmbeddingRequest?